ksql-common/src/main/java/io/confluent/ksql/function/KsqlFunction.java

```diff
@@ -188,7 +192,7 @@ public boolean equals(final Object o) {
       return false;
     }
     final KsqlFunction that = (KsqlFunction) o;
-    return Objects.equals(returnType, that.returnType)
+    return Objects.equals(returnSchemaProvider, that.returnSchemaProvider)
```

Comment:
I'm not sure that this will ever be true... if we need this support in `equals` and `hashCode` then we need to make a class out of the function, which would make things much more difficult. I think there are one of three things we can do:
- use the "java" return type because we know that at least it is somewhat accurate (e.g. BigDecimal will be some type of Decimal)
- completely remove the hashCodes/equals functionality. I don't think we need it anywhere in the code but we should double check that
- remove this check from hashCode and equals. This should be OK because it is impossible for two methods to have the same name and parameters but a different return type in java.

My preferred approach is the first, but then we have to pass in that "javaReturnType" as well, which feels a little hacky. This also would solve your toString issue. We should add unit tests for this (hashCode/equals) if it's important.